src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/Utilities/DeviceTest.cs

```diff
@@ -61,6 +98,16 @@ protected static void UnInstallTestApp ()
 			}
  		}
 
+		public static void RestartDevice (string project)
+		{
+			TestContext.Out.WriteLine ($"Trying to restart Emulator");
+			// shell out to msbuild and start the emulator again
+			using (var builder = new Builder ()) {
+				var out1 = RunProcessWithExitCode (builder.BuildTool, $"{project} /restore /t:AcquireAndroidTarget", timeoutInSeconds: 120);
```

Comment:
@dellis1972 : should `{project}` be *quoted* in case it's in a directory that contains spaces? e.g. `$"\"{project}\" /restore …"`?

Reply:
Well its not a path. We just pass the actual name of the project because it is in the same directory, so I don't think we need to

Reply:
This test doesn't create a project filename that contains spaces? https://github.com/xamarin/xamarin-android/blob/054a00c27563835d82d8b9eefc3c9dfdb8e1472e/src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/BuildTest.cs#L2498

Though that test isn't an on-device test, so it's not specifically relevant, just "example" relevant…